### PR TITLE
[UXIT-1793] - Ecosystem Explorer Epic · Rework Categories/Subcategories structure

### PR DESCRIPTION
**Pull request by CharlyMartin**
> [!IMPORTANT]
> This PR will only be merged when the feature is fully complete. It is a feature branch for this [effort](https://www.notion.so/filecoin/FF-Ecosystem-Explorer-Rework-Categories-Subcategories-structure-14b7631f2825807e9ed6d87fef734eb9?pvs=4). See https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/906#issuecomment-2539694833

## 📌 To-Do Before Merging
- [x] Update ecosystem project entries added after we started this rework to make `subcategory` the `category`

### CODE REVIEW

File: public/admin/config.yml [hunk 1]
```diff
@@ -567,25 +524,138 @@ collections:
...
-        value_field: "slug"
-        search_fields:
-          - name
-        display_fields:
-          - name
+        widget: select
+        options:
+          - label: "Artificial Intelligence"
+            value: "artificial-intelligence"
+            subcategories: ["ai-productivity-utilities"]
```
Comment (mirhamasala) [inline public/admin/config.yml:531]: What's is the `subcategories` field for?

File: public/admin/config.yml [hunk 2]
```diff
@@ -567,25 +524,138 @@ collections:
...
-          - name
-        filter:
-          field: "parent_category"
-          value: "{{fields.category}}"
+        widget: select
+        options:
+          # Artificial Intelligence subcategories
+          - label: "AI Productivity & Utilities"
+            value: "ai-productivity-utilities"
+            parent_category: "artificial-intelligence"
```
Comment (mirhamasala) [inline public/admin/config.yml:578]: What is the `parent_category` field for? These should be removed in this PR, right?

**Review by mirhamasala — CHANGES_REQUESTED** [review]
@CharlyMartin Please review the comments! Thank you. 🙏🏼

File: src/app/ecosystem-explorer/utils/getEcosystemCMSCategories.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { getCMSFieldOptions, getCollectionConfig } from '@/utils/cmsConfigUtils'
+
+import { splitCategoryAndGroup } from './splitCategoryAndGroup'
+
+export function getEcosystemCMSCategories() {
+  const { fields } = getCollectionConfig('ecosystem_projects')
+  const categories = getCMSFieldOptions(fields, 'category')
```
Comment (mirhamasala) [inline src/app/ecosystem-explorer/utils/getEcosystemCMSCategories.ts:7]:
```suggestion
  const categoryOptions = getCMSFieldOptions(fields, 'category')
```

File: src/app/ecosystem-explorer/utils/getEcosystemCMSCategories.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+  const categoriesWithGroup = categories.map((categoryOption) => {
```
Comment (mirhamasala) [inline src/app/ecosystem-explorer/utils/getEcosystemCMSCategories.ts:9]:
```suggestion
  const categoriesWithGroup = categoryOptions.map((categoryOption) => {
```

File: src/app/ecosystem-explorer/utils/getEcosystemCMSCategories.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    const { category, group } = splitCategoryAndGroup(categoryOption.label)
```
Comment (mirhamasala) [inline src/app/ecosystem-explorer/utils/getEcosystemCMSCategories.ts:10]:
```suggestion
    const { categoryGroup, categoryLabel } = splitCategoryGroupAndLabel(categoryOption.label)
```

File: src/app/ecosystem-explorer/utils/getEcosystemCMSCategories.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+    return {
+      value: categoryOption.value,
+      label: category,
```
Comment (mirhamasala) [inline src/app/ecosystem-explorer/utils/getEcosystemCMSCategories.ts:14]:
```suggestion
      label: categoryLabel,
```

File: src/app/ecosystem-explorer/utils/splitCategoryAndGroup.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { allowedCategoryGroups } from '../constants/allowedCategoryGroup'
+
+const SEPARATOR = ' - '
+
+export function splitCategoryAndGroup(label: string) {
```
Comment (mirhamasala) [inline src/app/ecosystem-explorer/utils/splitCategoryAndGroup.ts:5]:
```suggestion
export function splitCategoryAndGroup(groupAndCategoryLabel: string) {
```

File: src/app/ecosystem-explorer/utils/splitCategoryAndGroup.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  if (!label.includes(SEPARATOR)) {
+    throw new Error(`Category label does not contain "${SEPARATOR}"`)
```
Comment (mirhamasala) [inline src/app/ecosystem-explorer/utils/splitCategoryAndGroup.ts:7]:
```suggestion
    throw new Error(`Group and category label does not contain "${SEPARATOR}"`)
```

File: src/app/ecosystem-explorer/utils/splitCategoryAndGroup.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  const [group, category] = label.split(SEPARATOR)
```
Comment (mirhamasala) [inline src/app/ecosystem-explorer/utils/splitCategoryAndGroup.ts:10]:
```suggestion
  const [groupLabel, categoryLabel] = groupAndCategoryLabel.split(SEPARATOR)
```

File: src/app/ecosystem-explorer/utils/splitCategoryAndGroup.ts [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+  if (!category) {
+    throw new Error(`Category is missing from "${label}"`)
```
Comment (mirhamasala) [inline src/app/ecosystem-explorer/utils/splitCategoryAndGroup.ts:13]:
```suggestion
    throw new Error(`Category label is missing from "${label}"`)
```

**Comment by CharlyMartin** [comment]
The ecosystem project form works fine https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/947

The build fails because it's merging into `main`, which doesn't contain these changes yet

**Pull request merged**